firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/metadata/RolloutAssignment.java

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.metadata;
+
+import com.google.firebase.encoders.annotations.Encodable;
+import java.util.Objects;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * model used in user metadata context which make rollout assignment serialize, deserialize easier
+ */
+@Encodable
+public class RolloutAssignment {
```

Comment:
Is the thinking that this will live in the Interop SDK (once we check that in)?

And wonder if it could use [AutoValue](https://github.com/google/auto/blob/main/value/userguide/index.md#in-your-value-class) to take care of the boilerplate?

Reply:
Yup, this can live in interop and can be used by both RC and Crash, added AutoValue annotation.